assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -309,6 +314,18 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
         await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
       }
 
+      if (currentBlock.name === 'core/media-text') {
+        const uploadedImage = await getImageDetails(id);
+        const updatedAttributes = updateMediaAndTextAttributes(uploadedImage, currentBlock);
+        await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
+      }
+
+      if (currentBlock.name === 'planet4-blocks/carousel-header') {
```

Comment:
I think this should be a `else if` statement instead of just `if`

Reply:
Note that this comment becomes irrelevant if the code is combined to remove duplications as suggested above

Reply:
@Osong-Michael following the previous refactor, I think these should be `else if` statements instead of just `if`